ui/src/hooks/useFileManage.ts

```diff
@@ -0,0 +1,138 @@
+import type { MessageApiInjection } from 'naive-ui/es/message/src/MessageProvider';
+import { IFileManage, IFileManageConnectData, IFileManageSftpFileItem } from '@/hooks/interface';
+import { useFileManageStore } from '@/store/modules/fileManage.ts';
+
+enum MessageType {
+  CONNECT = 'CONNECT',
+  CLOSE = 'CLOSE',
+  ERROR = 'ERROR',
+  PING = 'PING',
+  PONG = 'PONG',
+  SFTP_DATA = 'SFTP_DATA'
+}
+
+/**
+ * @description 获取文件管理的 url
+ */
+const getFileManageUrl = () => {
+  const route = useRoute();
+
+  const routeName: RouteRecordNameGeneric = route.name;
+  const urlParams: URLSearchParams = new URLSearchParams(window.location.search.slice(1));
+
+  let fileConnectionUrl: string = '';
+
+  if (routeName === 'Terminal') {
+    fileConnectionUrl = urlParams ? `${BASE_WS_URL}/koko/ws/sftp/?${urlParams.toString()}` : '';
+
+    return fileConnectionUrl;
+  }
+};
+
+/**
+ * @description 处理 type 为 connect 的方法
+ * @param messageData
+ * @param id
+ * @param socket
+ */
+const handleSocketConnectEvent = (messageData: IFileManageConnectData, id: string, socket: WebSocket) => {
+  const sendData = '';
+
+  const sendBody = {
+    id,
+    type: 'SFTP_DATA',
+    cmd: 'list',
+    data: sendData
+  };
+
+  if (messageData) {
+    socket.send(JSON.stringify(sendBody));
+  }
+};
+
+const handleSocketSftpData = (messageData: IFileManageSftpFileItem[]) => {
+  const fileManageStore = useFileManageStore();
+
+  fileManageStore.setFileList(messageData);
+};
+
+/**
+ * @description 处理 message
+ * @param socket
+ */
+const initSocketEvent = (socket: WebSocket) => {
+  socket.binaryType = 'arraybuffer';
+
+  socket.onopen = () => {};
+  socket.onerror = () => {};
+
+  socket.onclose = (event: CloseEvent) => {};
+  socket.onmessage = (event: MessageEvent) => {
+    const message: IFileManage = JSON.parse(event.data);
+
+    switch (message.type) {
+      case MessageType.CONNECT: {
+        handleSocketConnectEvent(JSON.parse(message.data), message.id, socket);
+        break;
+      }
+
+      case MessageType.SFTP_DATA: {
+        handleSocketSftpData(JSON.parse(message.data));
+        break;
+      }
+
+      default: {
+        break;
+      }
+    }
+  };
+};
+
+/**
+ * @description 文件管理中的 Socket 连接
+ * @param url
+ * @param message
+ */
+const fileSocketConnection = (url: string, message: MessageApiInjection) => {
+  const { data, status, close, open, send, ws } = useWebSocket(url, {
+    protocols: ['JMS-KOKO'],
+    autoReconnect: {
+      retries: 5,
+      delay: 3000
+    }
+  });
+
+  if (!ws.value) {
+    message.error('获取文件列表信息失败');
+  }
+
+  initSocketEvent(<WebSocket>ws!.value);
+
+  return;
+};
+
+/**
+ * @description 用于处理文件管理相关逻辑
+ */
+export const useFileManage = () => {
+  let fileConnectionUrl: string | undefined = '';
+
+  const message = useMessage();
+
+  function init() {
+    fileConnectionUrl = getFileManageUrl();
+
+    if (fileConnectionUrl) {
+      fileSocketConnection(fileConnectionUrl, message);
+    }
+  }
+
+  init();
+};
```

Comment:
I checked the provided code with respect to its current state. There are no immediate issues or significant bugs.

However, some of the comments appear out-of-date:

```javascript
/** @description 获取文件管理的 url */
function getFileManageUrl() {

}

// The rest of the comment is missing

``` This line has been commented out. Please keep them updated as necessary based on future requirements or changes in design.

Overall, there are no apparent irregularities but you should always review old versions for compatibility reasons and consider updating when required.